scripts/generate-install-yaml.sh

```diff
@@ -27,3 +27,31 @@ kubebuilder create config --controller-image "${IMAGE_NAME}" \
 # Increase memory request and limit to avoid OOM issues.
 sed -i 's/memory: 20Mi/memory: 64Mi/' "${INSTALL_YAML}"
 sed -i 's/memory: 30Mi/memory: 128Mi/' "${INSTALL_YAML}"
+
+# Remove namespace resource from the top of the file
+sed -i -e '/---/,$!d' "${INSTALL_YAML}"
+
+# Remove namespace fields
+sed -i -e '/^  namespace: federation-system$/d' "${INSTALL_YAML}"
+
+if [ "${NS}" ]; then
+  # Add namespace to ClusterRoleBinding
+  sed -i -e '/- kind: ServiceAccount/a\
+  namespace: '${NS}'' "${INSTALL_YAML}"
+fi
+
+# Add args to container
+sed -i -e '/--install-crds=false/a\
+        - --federation-namespace=$(FEDERATION_NAMESPACE)' "${INSTALL_YAML}"
+
+# Add namespace env args to container
+sed -i -e '/terminationGracePeriodSeconds/i\
```

Comment:
Why is it desirable to add these args to the non-namespaced install yaml?

Reply:
As we need `FEDERATION_NAMESPACE` to be specified in L45, so that when a cluster join federation controller, it can join to different namespaces and federation controller can get those federationclusters from this namespace.

We do not need `CLUSTER_REGISTRY_NAMESPACE` here, will move it to `generate-namespaced-install-yaml.sh`.

Reply:
The only reason for these args is to allow the controller args to be set by the downward api.  In the non-namespaced case, `--federation-namespace` can be hardcoded (e.g. `--federation-namespace=${NS}`).

Reply:
Done.

One question @marun , can you please show some use case for the downward api? Why we need this for namespace scoped federation controller? Thanks.

Reply:
When the control plane is deployed namespaced, the downward api can be used to retrieve the namespace that the control manager pod is running in so the sync controllers can be configured to target that namespace.  Without this capability, it would not be possible for static yaml to deploy a namespaced control plane to an arbitrary namespace.